streams/src/main/java/org/apache/kafka/streams/state/internals/Segments.java

```diff
@@ -185,12 +186,31 @@ private void cleanup(final long segmentId) {
         }
     }
 
-    private long segmentIdFromSegmentName(String segmentName) {
-        try {
-            Date date = formatter.parse(segmentName.substring(name.length() + 1));
-            return date.getTime() / segmentInterval;
-        } catch (Exception ex) {
-            return -1L;
+    private long segmentIdFromSegmentName(String segmentName, final File parent) {
+        // old style segment name with date
+        if (segmentName.charAt(name.length()) == '-') {
+            final String datePart = segmentName.substring(name.length() + 1);
+            final Date date;
+            try {
+                date = formatter.parse(datePart);
+            } catch (ParseException e) {
+                throw new ProcessorStateException("Unable to parse segmentName:"
+                                                          + segmentName
+                                                          + " to a date");
+            }
+            final long segmentId = date.getTime() / segmentInterval;
+            final File newName = new File(parent, segmentName(segmentId));
+            final File oldName = new File(parent, segmentName);
+            if (!oldName.renameTo(newName)) {
+                throw new ProcessorStateException("Unable to rename old style segment from: "
+                                                          + oldName
+                                                          + " to new name: "
+                                                          + newName);
+            }
+            return segmentId;
+        } else {
+            return Long.parseLong(segmentName.substring(name.length() + 1));
```

Comment:
Should we catch a parsing exceptions and throw `ProcessorStateException` instead?